### PR TITLE
aws-accounts | update glue data catalog encryption

### PR DESCRIPTION
**Pull request by PratMis**
Adds support to update glue data catalog encryption.

### CODE REVIEW

File: c7n/resources/account.py [hunk 1]
```diff
@@ -1453,3 +1453,45 @@ def get_target(self, resource):
...
+                    attributes:
+                        DataCatalogEncryptionSettings:
+                            EncryptionAtRest:
+                                CatalogEncryption: SSE-KMS
+                                SseAwsKmsKeyId: alias/aws/glue
+    """
+
+    schema = type_schema(
+        'update-glue-datacatalog-encryption',
+        attributes={'type': 'object'},
```
Comment (kapilt) [inline c7n/resources/account.py:1484]: specify min attributes count on object

File: c7n/resources/account.py [hunk 2]
```diff
@@ -1453,3 +1453,45 @@ def get_target(self, resource):
...
+            policies:
+              - name: data-catalog-encryption
+                resource: account
+                filters:
+                  - type: glue-security-config
+                    CatalogEncryptionMode: DISABLED
+                actions:
+                  - type: update-glue-datacatalog-encryption
+                    attributes:
+                        DataCatalogEncryptionSettings:
```
Comment (kapilt) [inline c7n/resources/account.py:1476]: remove the redundant settings wrapper .. using pass through validation doesn't mean not thinking about usability.

File: c7n/resources/account.py [hunk 3]
```diff
@@ -1453,3 +1453,45 @@ def get_target(self, resource):
                 return []
             resource[self.annotation][kmskey] = self.data[kmskey]
         return resource[self.annotation]
+
+
+@actions.register('update-glue-datacatalog-encryption')
```
Comment (kapilt) [inline c7n/resources/account.py:1458]: rename set-glue-catalog-encryption

**Comment by PratMis** [comment]
Thanks for the feedback @kapilt, useful as always!! I have addressed your comments.

**Comment by PratMis** [comment]
Looks like there is macos ci/test failure unrelated to this PR. Traceback:
```
Traceback (most recent call last):
  File "/Users/runner/runners/2.168.0/work/cloud-custodian/cloud-custodian/.tox/py36/lib/python3.6/site-packages/pip/_internal/cli/base_command.py", line 186, in _main
    status = self.run(options, args)
  File "/Users/runner/runners/2.168.0/work/cloud-custodian/cloud-custodian/.tox/py36/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 404, in run
    use_user_site=options.use_user_site,
  File "/Users/runner/runners/2.168.0/work/cloud-custodian/cloud-custodian/.tox/py36/lib/python3.6/site-packages/pip/_internal/req/__init__.py", line 64, in install_given_reqs
    auto_confirm=True
  File "/Users/runner/runners/2.168.0/work/cloud-custodian/cloud-custodian/.tox/py36/lib/python3.6/site-packages/pip/_internal/req/req_install.py", line 686, in uninstall
    uninstalled_pathset = UninstallPathSet.from_dist(dist)
  File "/Users/runner/runners/2.168.0/work/cloud-custodian/cloud-custodian/.tox/py36/lib/python3.6/site-packages/pip/_internal/req/req_uninstall.py", line 544, in from_dist
    '(at %s)' % (link_pointer, dist.project_name, dist.location)
AssertionError: Egg-link /Users/runner/runners/2.165.2/work/cloud-custodian/cloud-custodian does not match installed location of c7n (at /Users/runner/runners/2.168.0/work/cloud-custodian/cloud-custodian)
```

**Comment by kapilt** [comment]
It’s more related to github actions changing afaics, I’ll investigate when I have a moment.

File: c7n/resources/account.py [hunk 4]
```diff
@@ -1453,3 +1453,46 @@ def get_target(self, resource):
...
+                resource: account
+                filters:
+                  - type: glue-security-config
+                    CatalogEncryptionMode: DISABLED
+                actions:
+                  - type: set-glue-catalog-encryption
+                    attributes:
+                        EncryptionAtRest:
+                            CatalogEncryption: SSE-KMS
+                            SseAwsKmsKeyId: alias/aws/glue
```
Comment (PratMis) [inline c7n/resources/account.py:1478]: Don' think using an alias directly is working in this case! There's no error that pops up but I think it needs a valid kms key arn! I'm investigating!

Comment (PratMis) [reply c7n/resources/account.py:1478]: Looks fine, works with alias! I have using a key that did not exist, which confused me. In the console, I saw an error  message asking for a valid key arn

Comment (PratMis) [reply c7n/resources/account.py:1478]: image
Tested by running a policy locally

**Comment by kapilt** [comment]
can you strip the response metadata?

**Comment by PratMis** [comment]
> can you strip the response metadata?

Absolutely!! I have taken out the metadata

**Pull request merged**